java/arcs/core/storage/handle/HandleManager.kt

```diff
@@ -0,0 +1,133 @@
+package arcs.core.storage.handle
+
+import arcs.core.crdt.CrdtSet
+import arcs.core.crdt.CrdtSingleton
+import arcs.core.data.CollectionType
+import arcs.core.data.EntityType
+import arcs.core.data.RawEntity
+import arcs.core.data.Schema
+import arcs.core.data.SingletonType
+import arcs.core.storage.ActivationFactory
+import arcs.core.storage.Callbacks
+import arcs.core.storage.ExistenceCriteria
+import arcs.core.storage.StorageKey
+import arcs.core.storage.StorageMode
+import arcs.core.storage.StorageProxy
+import arcs.core.storage.Store
+import arcs.core.storage.StoreOptions
+import arcs.core.storage.driver.RamDiskStorageKey
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+
+typealias SingletonData<T> = CrdtSingleton.Data<T>
+typealias SingletonOp<T> = CrdtSingleton.IOperation<T>
+typealias SingletonStoreOptions<T> = StoreOptions<SingletonData<T>, SingletonOp<T>, T?>
+typealias SingletonStore<T> = Store<SingletonData<T>, SingletonOp<T>, T?>
+typealias SingletonHandle<T> = SingletonImpl<T>
+typealias SingletonActivationFactory<T> = ActivationFactory<SingletonData<T>, SingletonOp<T>, T?>
+
+typealias SetData<T> = CrdtSet.Data<T>
+typealias SetOp<T> = CrdtSet.IOperation<T>
+typealias SetStoreOptions<T> = StoreOptions<SetData<T>, SetOp<T>, Set<T>>
+typealias SetStore<T> = Store<SetData<T>, SetOp<T>, Set<T>>
+typealias SetHandle<T> = CollectionImpl<T>
+typealias SetActivationFactory<T> = ActivationFactory<SetData<T>, SetOp<T>, Set<T>>
+
+interface ActivationFactoryFactory {
+    fun singletonFactory(): SingletonActivationFactory<RawEntity>
+    fun setFactory(): SetActivationFactory<RawEntity>
+}
+
+/**
+ * [HandleManager] is a convenience for creating handles using a provided store factory.
+ *
+ * Handles that are used by end-users will deal with [RawEntity], so this helper only bothers to
+ * create those types.
+ *
+ * It will create a [StorageProxy] for each new storage key, and keep a reference to it for as
+ * long as the [HandleFactory] exists.
+ *
+ * Instantiate it with the context and lifecycle that should own the resulting activate stores.
```

Comment:
This comment should probably go on the `Android` version's KDoc, rather than the core version's.